pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -30,13 +33,67 @@
 import org.apache.pinot.common.request.Literal;
 import org.apache.pinot.common.request.PinotQuery;
 import org.apache.pinot.spi.utils.BytesUtils;
+import org.apache.pinot.spi.utils.CommonConstants;
 import org.apache.pinot.sql.FilterKind;
+import org.apache.pinot.sql.parsers.CalciteSqlParser;
+import org.apache.pinot.sql.parsers.SqlCompilationException;
+import org.apache.pinot.sql.parsers.SqlNodeAndOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 public class RequestUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RequestUtils.class);
+
   private RequestUtils() {
   }
 
+  public static SqlNodeAndOptions parseQuery(String query, JsonNode request)
+      throws SqlCompilationException {
+    long parserStartTimeNs = System.nanoTime();
+    SqlNodeAndOptions sqlNodeAndOptions = CalciteSqlParser.compileToSqlNodeAndOptions(query);
+    setOptions(sqlNodeAndOptions, request);
+    sqlNodeAndOptions.setParseTimeNs(System.nanoTime() - parserStartTimeNs);
+    return sqlNodeAndOptions;
+  }
+
+  /**
+   * Sets extra options for the given query.
+   */
+  @VisibleForTesting
+  public static void setOptions(SqlNodeAndOptions sqlNodeAndOptions, JsonNode jsonRequest) {
+    Map<String, String> queryOptions = new HashMap<>();
+    if (jsonRequest.has(CommonConstants.Broker.Request.DEBUG_OPTIONS)) {
+      Map<String, String> debugOptions = RequestUtils.getOptionsFromJson(jsonRequest,
+          CommonConstants.Broker.Request.DEBUG_OPTIONS);
+      // TODO: remove debug options after releasing 0.11.0.
+      if (!debugOptions.isEmpty()) {
+        // NOTE: Debug options are deprecated. Put all debug options into query options for backward compatibility.
```

Comment:
This is not enough. During version upgrade, server doesn't know the debug options can be read from the query options